gcsio/src/test/java/com/google/cloud/hadoop/gcsio/integration/GoogleCloudStorageImplTest.java

```diff
@@ -209,6 +209,62 @@ public void testCopySingleItemWithRewrite() throws IOException {
     assertObjectContent(gcs, copiedResourceId, partition, partitionsCount);
   }
 
+  @Test
+  public void testUploadAndGetObjectWithEncryptionKey() throws IOException {
+    GoogleCloudStorageImpl gcs =
+        makeStorage(
+            GoogleCloudStorageTestHelper.getStandardOptionBuilder()
+                .setEncryptionAlgorithm("AES256")
+                .setEncryptionKey("CSX19s0epGWZP3h271Idu8xma2WhMuKT8ZisYfcjLM8=")
+                .setEncryptionKeyHash("LpH4y6Bki5zIhYrjGo1J4BuSt12G/1B+n3FwORpdoyQ=")
+                .build());
+
+    String bucketName = BUCKET_HELPER.getUniqueBucketName("encryption-bucket");
+    StorageResourceId resourceId = new StorageResourceId(bucketName, "obj");
+
+    gcs.create(bucketName);
+    gcs.createEmptyObject(
+        resourceId,
+        new CreateObjectOptions(true, "text/plain", CreateObjectOptions.EMPTY_METADATA));
+
+    assertThat(gcs.getItemInfo(resourceId).getContentType()).isEqualTo("text/plain");
+  }
+
+  @Test
+  public void testRewriteEncryptedObject() throws IOException {
+    GoogleCloudStorageImpl gcs =
+        makeStorage(
+            GoogleCloudStorageTestHelper.getStandardOptionBuilder()
+                .setCopyWithRewriteEnabled(true)
+                .setMaxBytesRewrittenPerCall(512 * 1024 * 1024)
+                .setEncryptionAlgorithm("AES256")
```

Comment:
May you extract encryption configuration in common method and reuse it in these tests?

Reply:
> > Added integrations to test insert/get/rewrite. right now hardcoded the encryption key (but functional). Might consider to add functions to generate key and hash automatically for each run.
> 
> Does it require any project or GCS bucket set-up? Or anybody can run these tests against any bucket in any project?

Unlike CMEK which requires a GCS key ring, CSEK doesn't need any preconfiguration as long as key and key hash match themseleves.

Reply:
Done